src/elfpy/utils/sim_utils.py

```diff
@@ -112,34 +112,65 @@ def get_init_lp_agent(
-    # budget is the full amount for LP & short
-    budget = first_base_to_lp + short_amount + second_base_to_lp
-    # construct the init_lp agent with desired budget, lp, and short amounts
-    init_lp_agent = import_module("elfpy.policies.init_lp").Policy(
+    # to hit the desired reserve ratio, we open a short which increases bond reserves exactly by the amount we want
+    # this short trade decreases share reserves by an exact amount that we know (delta_shares)
+    # since this is undesired, we add delta_shares to the initial liquidity amount (first_base_lp)
+    # however, this new initial LP amount changes the result of the short trade due to a change in slippage
+    # this change in slippage is relatively small, and added to the estimate at each iteration, converges quickly
+    # for detailed convergence process: https://github.com/element-fi/elf-simulations/pull/136#issuecomment-1405922764
+    delta_shares = seed_liquidity
+    prev_delta_shares = 0
+    trade_result = None
+    iteration = 0
+    absolute_tolerance = 1e-20
+    while abs(delta_shares - prev_delta_shares) > absolute_tolerance and iteration < 20:
+        iteration += 1
+        trade_result = market.pricing_model.calc_out_given_in(
+            in_=Quantity(amount=init_bond_reserves, unit=TokenType.PT),
+            market_state=MarketState(
+                share_reserves=init_share_reserves + delta_shares,  # what will turn into first_base_lp (mock-up)
+                bond_reserves=0,
+                share_price=market.market_state.share_price,
+                init_share_price=market.market_state.init_share_price,
+            ),
+            fee_percent=fee_percent,
+            time_remaining=market.position_duration,
+        )
+        prev_delta_shares = delta_shares
+        delta_shares = trade_result.user_result.d_base / market.market_state.share_price
+        logging.debug(
+            (
+                "\niteration %g: init_share_reserves=%g delta_shares=%g"
+                "\na relative change of %.2e"
+                "\nestimate for first_base_lp: %g"
+            ),
+            iteration,
+            init_share_reserves,
+            delta_shares,
+            (delta_shares - prev_delta_shares) / prev_delta_shares,
+            (init_share_reserves + delta_shares) * market.market_state.share_price,
+        )
+    logging.debug(
+        ("\nstopping at iteration %g because delta_delta_shares is <= %g"),
+        iteration,
+        absolute_tolerance,
+    )
+    delta_base = delta_shares * market.market_state.share_price
+    max_loss_in_base = init_bond_reserves - delta_base
+    init_base_reserves = init_share_reserves * market.market_state.share_price
+    first_base_to_lp = init_base_reserves + delta_base  # add back delta_base to immunize effect of short
+    second_base_to_lp = target_liquidity - init_base_reserves  # remember delta_base is immunized, doesn't need adding
```

Comment:
shouldn't this be 
```second_base_to_lp = target_liquidity - first_base_to_lp - max_loss_in_base```

Reply:
the first action by the agent should send `first_base_to_lp` to the market and the second action should send `max_loss_in_base` to the market no?

Reply:
if the liquidity adds up correctly then something is wrong in our market accounting.  Let's add a TODO if that's the case and follow up with another PR.

Reply:
the second sends `max_loss_in_base` into the protocol, but keeps it in the margin account outside of the pool reserves

Reply:
don't need to use `first_base_to_lp` because the end result of the first LP's accounts is `init_base_reserves` in the pool, not `first_base_to_lp` which is made up of `init_base_reserves + delta_base`. the first LP does put both components into the market, but the second piece (`delta_base`) is there only to offset the effect of the short, so after the short, the market holds `init_base_reserves` base.

Reply:
let me know if you think there's something missing to this logic, we can add the TODO and/or investigate further. gonna resolve and merge for now.

Reply:
yeah but `max_loss_in_base` still increases the total liquidity of the protocol.  the smart contract will have that much more base flow into it and the agent will spend that much base, so we should account for it if we aren't.  Add a TODO to investigate.

Reply:
added